unidist/core/backends/mpi/core/worker/request_store.py

```diff
@@ -110,21 +115,27 @@ def check_pending_get_requests(self, data_ids):
         data_id : iterable or unidist.core.backends.common.data_id.DataID
             An ID or list of IDs to data.
         """
+
+        def check_data_id(data_id):
+            if data_id in self._data_request:
+                ranks_with_get_request = self._data_request[data_id]
+                for rank_num in ranks_with_get_request:
+                    # Data is already in DataMap, so not problem here
+                    self.process_get_request(rank_num, data_id, is_blocking_op=False)
+                del self._data_request[data_id]
+            #
+            if data_id in self._blocking_data_request:
```

Comment:
```suggestion
            elif data_id in self._blocking_data_request:
```

Reply:
The data_id can be requested by both a blocking request and a non-blocking request, so we need to check in both lists.